src/quo2/components/dropdown.cljs

```diff
@@ -0,0 +1,150 @@
+                                                      :padding-bottom 5
+                                                      :padding-left 12
+                                                      :padding-right 8}}}
+            :small {:icon-size 12
+                    :font {:font-size :label}
+                    :height 24
+                    :padding {:padding-with-icon {:padding-vertical 3
+                                                  :padding-left 6
+                                                  :padding-right 6}
+                              :padding-with-no-icon {:padding-vertical 3
+                                                     :padding-horizontal 8}}}})
+(defn color-by-10
+  [color]
+  (colors/alpha color 0.6))
+
+(defn dropdown-comp [{:keys [icon open? text dd-height size disabled? dd-color border-color]}]
+  (let [dark? (colors/dark?)
+        {:keys [width height width-with-icon padding font icon-size]} (get-in sizes [size])
+        {:keys [padding-with-icon padding-with-no-icon]} padding
+        font-size (:font-size font)
+        spacing 4]
+    [rn/touchable-opacity (cond-> {:on-press (fn []
+                                               (if (swap! open? not)
+                                                 (apply-anim dd-height 120)
+                                                 (apply-anim dd-height 0)))
+                                   :style (cond-> (merge
+                                                   (if icon
+                                                     padding-with-icon
+                                                     padding-with-no-icon)
+                                                   {:width (if icon
+                                                             width-with-icon
+                                                             width)
+                                                    :height height
+                                                    :border-radius (if (= :small size)
+                                                                     8
+                                                                     10)
+                                                    :flex-direction :row
+                                                    :align-items :center
+                                                    :border-width 1
+                                                    :border-color (if @open?
+                                                                    border-color
+                                                                    (color-by-10 border-color))
+                                                    :background-color (if @open?
+                                                                        dd-color
+                                                                        (color-by-10 dd-color))})
+                                            disabled? (assoc :opacity 0.3))}
```

Comment:
I had a bit of difficulty reading the nested `cond->` calls because it was hard to keep track of which map was being updated (i.e. the inner style map or the parent one). In case you like my like suggestion, here's the code:

```clojure
(cond-> {:on-press (fn []
                     (if (swap! open? not)
                       (apply-anim dd-height 120)
                       (apply-anim dd-height 0)))
         :style (merge
                 (if icon
                   padding-with-icon
                   padding-with-no-icon)
                 {:width (if icon
                           width-with-icon
                           width)
                  :height height
                  :border-radius (if (= :small size)
                                   8
                                   10)
                  :flex-direction :row
                  :align-items :center
                  :border-width 1
                  :border-color (if @open?
                                  border-color
                                  (color-by-10 border-color))
                  :background-color (if @open?
                                      dd-color
                                      (color-by-10 dd-color))})}
  disabled? (assoc-in [:style :opacity] 0.3)
  disabled? (assoc :disabled true))
```

Reply:
Definitely more readable .. Thanks